src/models/CheckoutParams.ts

```diff
@@ -16,7 +16,7 @@ interface CheckoutParams {
 
     note?: string;
     
-    tipSettings?:TipSettings;
+    tipSettings:TipSettings;
```

Comment:
This should be optional per this doc: https://developer.squareup.com/docs/api/reader/ios/. 

Does this throw on iOS or Android?

Reply:
Actually No i'm not getting any exception with this.

Reply:
Do you mean tipSettings can be "undefined" and no exception throw?  if so, can you please check with @caleb-harrelson in this comment https://github.com/square/react-native-square-reader-sdk/pull/148#discussion_r987182878 ?

Reply:
> Do you mean tipSettings can be "undefined" and no exception throw? 

The changes in this PR include the changes in #198, so the answer here would be yes. 

I still maintain that the correct remedy is #199, which reverts the entire PR that “added” TypeScript in a partial implementation that does not expose the types and contains more inaccuracies than were fixed herein.